specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json

```diff
@@ -2025,6 +2239,42 @@
       },
       "description": "AmlCompute update parameters."
     },
+    "NodeUserAccountsInformation": {
+      "description": "Compute node's user accounts information.",
+      "type": "object",
+      "properties": {
+        "nodesUserAccountsInformation": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NodeUserAccountInformation"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The continuation token."
+        }
+      }
+    },
+    "NodeUserAccountInformation": {
```

Comment:
Since this is returned from get .../users, this type should be modeled as a resource with id, name, type, properties...

Would it make more sense to just have a get /computes/{computeName}/nodeUsers api which returns nodeUserCollection. Think of this as a higher level collection enumeration of nodeUser (similar to how you can get top level resources at sub or rg scope). This wouldn't automatically bulk user by nodeId but may be a cleaner API. If you want to bulk them together like this you could call this post /computes/{computeName}/listNodeUsers and have a payload shape consistent with whats currently in the PR.